docs/types.rst

```diff
@@ -266,6 +266,118 @@ to and from all integer types but implicit conversion is not allowed.
         }
     }
 
+.. index:: ! function type, ! type; function
+
+.. _function_types:
+
+Function Types
+--------------
+
+Functions can be assigned to variables and passed on together with function calls.
```

Comment:
I want another sentence here.  `The variables and the function arguments that contain functions have function types.  A function type specifies the argument types and the return type of the function.  Moreover (continues next line)`